### PR TITLE
Include more use cases

### LINKED ISSUE #263: Complete use cases in DG
**Description**
Complete the use cases in DG

### PR DESCRIPTION
**Pull request by LynetteLeeSiYing**
**Description**
Include important use cases into developer guide.

close #263

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -332,33 +332,77 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index or name is invalid.
+  - 3a1. AddressBook shows an error message.
```
Comment (Bonzzz3) [inline docs/DeveloperGuide.md:257]: Should be MediContact here

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -332,33 +332,77 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  
+    Use case resumes at step 2.
+  
+* 4a. User chooses to cancel the deletion.
+
+  Use case ends.
+
+**Use case: Clear address book**
```
Comment (Bonzzz3) [inline docs/DeveloperGuide.md:265]: I think "Clear all contacts in MediContact" might be a better way to phrase this.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -332,33 +332,77 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**MSS**
+
+1.  User requests to clear address book.
```
Comment (Bonzzz3) [inline docs/DeveloperGuide.md:269]: Same as above

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -368,42 +412,174 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-  - 2b1.Medicontact shows an error message specifying the incorrect format.
-  - 2b2. Use case ends.
-- 2c. No match in users
-  - 2b1.Medicontact shows an error message specifying that there are no users that match the query.
-  - 2b2. Use case ends.
+  - 2b1.MediContact shows an error message specifying the incorrect format.
+  
+    Use case ends.
+  
+**Use case: Import address book**
```
Comment (Bonzzz3) [inline docs/DeveloperGuide.md:356]: "Imports contact info from the given json file into MediContact." This is from our DG. Maybe can try update the use case closer to this?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -368,42 +412,174 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**MSS**
+
+1. User requests to import address book.
+2. User inputs the import command with new address book file name.
```
Comment (Bonzzz3) [inline docs/DeveloperGuide.md:361]: See above

**Review by Bonzzz3 — COMMENTED** [review]
Logic of all test cases LGTM! Just some phrases that could be improved. Thankss

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -216,35 +216,77 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-* 2a. Necessary field is missing
+* 2a. Necessary field is missing.
+    * 2a1. MediContact shows an error message indicating which field is missing.
+  
+      Use case ends.
 
-    * 2a1. Medicontact shows an error message indicating which field is missing
-    * 2a2. Use case ends.
+* 2b. Wrong format in input.
+    * 2b1.MediContact shows an error message specifying the incorrect format.
```
Comment (myrilla12) [inline docs/DeveloperGuide.md:233]: Missing whitespace in '2b1.MediContact'

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -269,30 +337,136 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  
+    Use case ends.
+  
 - 2b. Wrong format in input
-  - 2b1.Medicontact shows an error message specifying the incorrect format.
-  - 2b2. Use case ends.
-- 2c. No match in users
-  - 2b1.Medicontact shows an error message specifying that there are no users that match the query.
-  - 2b2. Use case ends.
+  - 2b1.MediContact shows an error message specifying the incorrect format.
```
Comment (myrilla12) [inline docs/DeveloperGuide.md:352]: Missing whitespace in '2b1.MediContact'

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -269,30 +337,136 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Extensions**
+
+- 2a. File is not in the same folder as application JAR file.
+    - 2a1. MediContact shows an error message indicating file not found.
+
+      Use case ends.
+
+- 2b. File is not in JSON format.
+    - 2b1.MediContact shows an error message indicating wrong file format.
```
Comment (myrilla12) [inline docs/DeveloperGuide.md:375]: Missing whitespace

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -269,30 +337,136 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+      Use case ends.
+  
+- 2c. File is not in expected format of MediContact data.
+    - 2c1.MediContact shows an error message indicating invalid JSON format.
```
Comment (myrilla12) [inline docs/DeveloperGuide.md:380]: Missing whitespace

**Review by myrilla12 — CHANGES_REQUESTED** [review]
Looks good! Just some very minor typos. Also a suggestion, since the UG is going to reference 'patient' instead of 'person' we can consider doing the same for the DG.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (LynetteLeeSiYing) [reply docs/DeveloperGuide.md:356]: Since in UG we state as 'Exporting the data' and 'Saving the data' , do you think 'Import data' suitable here?

**Review by ritvi12 — APPROVED** [review]
LGTM!

**Review by OttoTan — APPROVED** [review]
LGTM

**Pull request merged**